### PR TITLE
Feature/global button

### PR DESCRIPTION
**Pull request by k-y0u**
Add a big save button for model view.

### CODE REVIEW

File: src/composables/events/FileEvent.js [hunk 1]
```diff
@@ -52,6 +52,12 @@ const ExpandFolderEvent = new Subject();
  */
 const UpdateFileEvent = new Subject();
 
+/**
+ * Represent a rxjs Event object to emit and to receive events file global saving.
```
Comment (justinehell) [inline src/composables/events/FileEvent.js:56]: `...to receive events about files global saving.`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/composables/Project.js [hunk 2]
```diff
@@ -589,3 +589,38 @@ export async function gitCommit(projectId, message) {
     message,
   });
 }
+
+export async function gitGlobalSave(project) {
```
Comment (Zorin95670) [inline src/composables/Project.js:593]: missing documentation

File: src/composables/Project.js [hunk 3]
```diff
@@ -589,3 +589,38 @@ export async function gitCommit(projectId, message) {
...
+    `leto-modelizer ${nowDate.toDateString()}`,
+  );
+
+  await gitPush(
+    project,
+    newBranch,
+    true,
+  );
+
+  FileEvent.GlobalSaveFilesEvent.next();
```
Comment (Zorin95670) [inline src/composables/Project.js:625]: to be compliant with other function please make a return here

File: src/components/ModelizerNavigationBar.vue [hunk 4]
```diff
@@ -65,7 +82,40 @@ const buttonToggleOptions = ref([{
...
+  return !project.value.git?.repository;
+}
+
+function getSavebuttonTitle() {
+  return isSaveButtonDisable()
+    ? 'page.modelizer.header.button.save.disable.title'
+    : 'page.modelizer.header.button.save.enable.title';
+}
+
+async function save() {
```
Comment (Zorin95670) [inline src/components/ModelizerNavigationBar.vue:97]: image

File: src/composables/Project.js (hunk 3 above)
Comment (k-y0u) [reply src/composables/Project.js:625]: useless but done

File: src/composables/events/FileEvent.js (hunk 1 above)
Comment (k-y0u) [reply src/composables/events/FileEvent.js]: done

File: src/composables/Project.js (hunk 2 above)
Comment (k-y0u) [reply src/composables/Project.js:597]: done

File: src/components/ModelizerNavigationBar.vue (hunk 4 above)
Comment (k-y0u) [reply src/components/ModelizerNavigationBar.vue:98]: done

File: tests/unit/components/ModelizerNavigationBar.spec.js [hunk 5]
```diff
@@ -86,6 +89,59 @@ describe('Test component: ModelizerNavigationBar', () => {
     });
   });
 
+  describe('Test computed: isSaveButtonDisable', () => {
+    it('should return false if project git repository is definied', () => {
```
Comment (Zorin95670) [inline tests/unit/components/ModelizerNavigationBar.spec.js:93]: definied ?

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: tests/unit/components/ModelizerNavigationBar.spec.js (hunk 5 above)
Comment (k-y0u) [reply tests/unit/components/ModelizerNavigationBar.spec.js]: done

**Review by justinehell — APPROVED** [review]

**Review by justinehell — DISMISSED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**